File: the-lambda-trilogy/python/the_lambda_trilogy/the_lambda_lith_stack.py
from aws_cdk import (
    aws_lambda as _lambda,
    aws_apigateway as api_gw,
    core
)


class TheLambdalithStack(core.Stack):

    def __init__(self, scope: core.Construct, id: str, **kwargs) -> None:
        super().__init__(scope, id, **kwargs)

        lambda_lith = _lambda.Function(self, "lambdalithHandler",
                                       runtime=_lambda.Runtime.PYTHON_3_8,
                                       handler="lambdalith.handler",
                                       code=_lambda.Code.from_asset("lambda_fns/the_lambda_lith/flask")
                                       )

        # defines an API Gateway REST API resource backed by our "lambda_lith" function.
        api_gw.LambdaRestApi(self, 'LambdalithAPI',
                             handler=lambda_lith
                             )

File: the-lambda-trilogy/typescript/lib/the-lambda-lith-stack.ts
import * as cdk from '@aws-cdk/core';
import lambda = require('@aws-cdk/aws-lambda');
import apigw = require('@aws-cdk/aws-apigateway');

export class TheLambdalithStack extends cdk.Stack {
  constructor(scope: cdk.Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, props);

    /**
     * Only one lambda, the lambdalith
     */

    const lambdalith = new lambda.Function(this, 'lambdalithHandler', {
      runtime: lambda.Runtime.NODEJS_12_X,
      code: lambda.Code.fromAsset('lambda-fns/the-lambda-lith'),
      handler: 'lambdalith.main',                
    });


    /**
     * All routes go to lambdalith which handles routes internally
     */
    
    new apigw.LambdaRestApi(this, 'LambdalithAPI', {
      handler: lambdalith
    });
  }
}
